### PR TITLE
[AMB-166][AMB-170] feature: Initial mailgun module

### PR DESCRIPTION
**Pull request by bufo24**

### CODE REVIEW

File: src/libs/mail/mail.service.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  async send({ subject, email, variables }: SendEmailProps) {
+    if (!this.mailgun) return;
+
+    const data: MailgunMessageData = {
+      from: `Banco <noreply@${this.mailgun.domain}>`,
+      to: [email],
+      subject,
+      template: '',
```
Comment (apotdevin) [inline src/libs/mail/mail.service.ts:44]: Would this work with an mjml template?

File: src/libs/mail/mail.service.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    client: IMailgunClient;
+    domain: string;
+  };
+
+  constructor(
+    private configService: ConfigService,
+    @Logger(MailService.name) private logger: CustomLogger,
+  ) {
+    const apiKey = this.configService.get<string>(`mailgun.apiKey`);
+    const domain = this.configService.get<string>(`mailgun.domain`);
```
Comment (apotdevin) [inline src/libs/mail/mail.service.ts:22]: Don't see this env in the env zod validation schema

File: src/libs/mail/mail.service.ts (hunk 1 above)
Comment (bufo24) [reply src/libs/mail/mail.service.ts:44]: There should the name go of the template, which is defined in Mailgun

Comment (apotdevin) [reply src/libs/mail/mail.service.ts:44]: Can we pass directly in code so that we don't need to add templates in mailgun?

Comment (bufo24) [reply src/libs/mail/mail.service.ts:44]: Yes we can also pass in directly HTML.

**Pull request merged**